backend/core/llm_chain.py

```
@@ -32,17 +33,62 @@
+    def __init__(self):
+        self._rate_limit_status: Dict[str, bool] = {}
+        self._rate_limit_reset_time: Dict[str, datetime] = {}
+
+    def is_rate_limited(self, provider: str) -> bool:
+        """Check if a provider is currently rate limited"""
+        if provider not in self._rate_limit_status:
+            return False
+
+        # Check if rate limit has expired
+        if provider in self._rate_limit_reset_time:
+            if datetime.now() > self._rate_limit_reset_time[provider]:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)